src/panels/lovelace/strategies/area/area-view-strategy.ts

```diff
@@ -0,0 +1,193 @@
+import { ReactiveElement } from "lit";
+import { customElement } from "lit/decorators";
+import type { LovelaceBadgeConfig } from "../../../../data/lovelace/config/badge";
+import type { LovelaceSectionRawConfig } from "../../../../data/lovelace/config/section";
+import type { LovelaceViewConfig } from "../../../../data/lovelace/config/view";
+import type { HomeAssistant } from "../../../../types";
+import type { EntitiesFilterSectionStrategyConfig } from "../entities-filter/entities-filter-section-strategy";
+
+export interface AreaViewStrategyConfig {
+  type: "area";
+  area?: string;
+}
+
+@customElement("area-view-strategy")
+export class AreaViewStrategy extends ReactiveElement {
+  static async generate(
+    config: AreaViewStrategyConfig,
+    hass: HomeAssistant
+  ): Promise<LovelaceViewConfig> {
+    if (!config.area) {
+      throw new Error("Area not provided");
+    }
+
+    const area = hass.areas[config.area];
+
+    if (!area) {
+      throw new Error("Unknown area");
+    }
+
+    const sections: LovelaceSectionRawConfig[] = [];
+
+    const badges: LovelaceBadgeConfig[] = [];
+
+    if (area.temperature_entity_id) {
+      badges.push({
+        entity: area.temperature_entity_id,
+        type: "entity",
+        color: "red",
+      });
+    }
+
+    if (area.humidity_entity_id) {
+      badges.push({
+        entity: area.humidity_entity_id,
+        type: "entity",
+        color: "indigo",
+      });
+    }
+
+    // Lights
+    sections.push({
+      strategy: {
+        type: "entities-filter",
+        title: "Lights",
+        icon: "mdi:lamps",
+        filter: {
+          domain: "light",
+          area: config.area,
+          entity_category: "none",
+        },
+      } as EntitiesFilterSectionStrategyConfig,
```

Comment:
Doesn't this prevent the data structure from being type checked?

Reply:
Right, we should use `satisfies` keyword.